saleor/graphql/notifications/tests/test_external_notification_dispatch.py

```diff
@@ -0,0 +1,187 @@
+from unittest.mock import patch
+
+from graphql_relay.node.node import to_global_id
+
+from ....core.notify_events import UserNotifyEvent
+from ....product.models import ProductVariant
+
+
+@patch("saleor.plugins.sendgrid.tasks.send_email_with_dynamic_template_id.delay")
```

Comment:
We shouldn't mock any plugins from Saleor. Think of a plugin as something that doesn't belong to Saleor. 
Here you should mock the proper call of the manager and call `assert_called_once_with(..)`. 
Then you can add tests in sendgrid module to confirm that sendgrid Plugin triggered proper action.